### PR TITLE
fix(registry): Handle currently impossible edge case in canister_ranges_ updates

### PR DESCRIPTION
**Pull request by max-dfinity**
This adds a test, and fixes the edge case where a new routing table could have a mutation that doesn't have an existing range to fall into.  This is currently not possible, but this change makes the code more robust so that it cannot unnecessarily panic if there's no range starting at CanisterId "0".

### CODE REVIEW

File: rs/registry/canister/src/mutations/routing_table.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ pub(crate) fn mutations_for_canister_ranges(
     let zero_id = CanisterId::from_u64(0);
     // If we don't have any routing table fragments in the new_shards, we need to create a default one to
     // hold all the ranges until a split occurs.
-    if new_shards.is_empty() {
+    if !new_shards.contains_key(&zero_id) {
```
Comment (jasonz-dfinity) [inline rs/registry/canister/src/mutations/routing_table.rs]: Nice. I believe this makes it obvious that the `.unwrap()` on L98 is impossible to fail. Perhaps document this?

**Review by jasonz-dfinity — APPROVED** [review]

File: rs/registry/canister/src/mutations/routing_table.rs (hunk 1 above)
Comment (max-dfinity) [reply rs/registry/canister/src/mutations/routing_table.rs]: done!
